lib/broadway_rabbitmq/amqp_client.ex

```diff
@@ -285,8 +285,11 @@ defmodule BroadwayRabbitMQ.AmqpClient do
   end
 
   defp maybe_declare_queue(channel, queue, declare_opts) do
-    with {:ok, %{queue: queue}} <- Queue.declare(channel, queue, declare_opts) do
-      {:ok, queue}
+    case Queue.declare(channel, queue, declare_opts) do
+      :ok -> {:ok, queue}
+      {:ok, _} -> {:ok, queue}
+      {:error, :ok} -> {:ok, queue}
```

Comment:
This error is not documented in the AMQP library though. When does it happen?

Reply:
with amqp at version `1.X`(I've tested it particularly with `1.6.0`) set option `no_wait: true`.

Reply:
I meant in what conditions does this happen? If you set `no_wait: true` it always returns `{:error, :ok}`?

Reply:
exactly

Reply:
It doesn't feel like we want to simply ignore it, but instead we want to raise. As said on #96:

> However, regardless the bug, I don't think you should use the option with Broadway.

We probably should raise whenever the option is given.

Reply:
Yes, agreed with @josevalim

Reply:
I don't see any issue with supporting this feature from a Broadway perspective. We can discuss that on #96. The `:no_wait` flag just makes the declaration and binding async, it doesn't impact subscribing the consumer to a queue. It just gonna make the start-up of the broadway producer faster.
If the biggest issue is the `{:error, :ok}` pattern match, it's just needed to enforce amqp to be version `2.X`.

Reply:
@ono is ampq maintainer and said the feature is not really encouraged here. What happens if we try to subscribe to a queue that does not exist? That's what can happens if we use nowait.

Reply:
@josevalim isn't the case to just add a warning for that in the documentation?
My use case is that I know for sure that the queue gonna exist at the point of starting broadway, that's why waiting synchronously for that confirmation isn't useful.

Reply:
If you know it exists, do you still need to declare it?

Reply:
I just realized that I misunderstood the docs on that part. I was understanding that if I don't have declare options it would use `""` as the queue name. I gonna update the docs on the part.